### PR TITLE
[SPARK-33140][SQL] remove SQLConf and SparkSession in all sub-class of Rule[QueryPlan]

### PR DESCRIPTION
**Pull request by leanken-zz**
### What changes were proposed in this pull request?

Since Issue [SPARK-33139](https://issues.apache.org/jira/browse/SPARK-33139) has been done, and SQLConf.get and SparkSession.active are more reliable. We are trying to refine the existing code usage of passing SQLConf and SparkSession into sub-class of Rule[QueryPlan].

In this PR.

* remove SQLConf from ctor-parameter of all sub-class of Rule[QueryPlan].
* using SQLConf.get to replace the original SQLConf instance.
* remove SparkSession from ctor-parameter of all sub-class of Rule[QueryPlan].
* using SparkSession.active to replace the original SparkSession instance.

### Why are the changes needed?

Code refine.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?

Existing test

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala [hunk 1]
```diff
@@ -28,7 +28,8 @@ import org.apache.spark.sql.types.{StructField, StructType}
 /**
  * An analyzer rule that replaces [[UnresolvedInlineTable]] with [[LocalRelation]].
  */
-case class ResolveInlineTables(conf: SQLConf) extends Rule[LogicalPlan] with CastSupport {
+object ResolveInlineTables extends Rule[LogicalPlan] with CastSupport {
+  override def conf: SQLConf = SQLConf.get
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala:32]: Instead, could we define `final def conf: SQLConf = SQLConf.get` in the base class `Rule` if this is a common pattern?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 2]
```diff
@@ -182,7 +182,7 @@ class Analyzer(
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:182]: Could we remove `conf: SQLConf` in the constructor of `Analyzer`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala (hunk 1 above)
Comment (leanken) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala:32]: If it's final could not be override, and in its subclass the usage difference would be conf.XXX and SQLConf.get.XXX, I don't think we should bother to add such interface.  ^_^

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala (hunk 2 above)
Comment (leanken) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:182]: > Could we remove `conf: SQLConf` in the constructor of `Analyzer`?

My original idea is to remove this parameter, but it's causing much more changing on the Test code. @cloud-fan , what's your thought?

**Review by viirya — COMMENTED** [review]
> remove SQLConf from ctor-parameter of all sub-class of Rule[QueryPlan].
> using SQLConf.get to replace the original SQLConf instance.

But even you clean this up in this PR, technically we still possibly do passing `SQLConf` in future development.

Is any other benefit from removing the SQLConf?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala (hunk 2 above)
Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:182]: I think it's fine to keep it. The main goal is to let rules use `SQLConf.get`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 3]
```diff
@@ -182,7 +182,7 @@ class Analyzer(
 
   private def executeSameContext(plan: LogicalPlan): LogicalPlan = super.execute(plan)
 
-  def resolver: Resolver = conf.resolver
+  def resolver: Resolver = SQLConf.get.resolver
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:185]: unnecessary change.

**Comment by leanken** [comment]
@cloud-fan @maropu test passed, could you please have a look? thanks ^_^

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala [hunk 4]
```diff
@@ -29,22 +29,9 @@ import org.apache.spark.sql.catalyst.rules.Rule
...
- * partitions will be kept in [[HiveTableRelation.prunedPartitions]], and the statistics of
- * the hive table relation will be updated based on pruned partitions.
- *
- * This rule is executed in optimization phase, so the statistics can be updated before physical
- * planning, which is useful for some spark strategy, eg.
- * [[org.apache.spark.sql.execution.SparkStrategies.JoinSelection]].
- *
- * TODO: merge this with PruneFileSourcePartitions after we completely make hive as a data source.
- */
-private[sql] class PruneHiveTablePartitions(session: SparkSession)
```
Comment (viirya) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: Why remove these comments?

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: When i change from class to object, the building task java unidoc failed。Do you know how to fix this issue??

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: -Phadoop-3.2 -Phive-2.3 -Phadoop-cloud -Pkinesis-asl -Pspark-ganglia-lgpl -Pyarn -Pkubernetes -Phive -Pmesos -Phive-thriftserver unidoc

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: I will upload the error log later to see if you can help me on this issue.

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: https://github.com/akka/akka-http/issues/1895
something might be relative??

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: error message as following

```
[error] /Users/linxuewei/Code/github/leanken-spark/sql/hive/target/java/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.java:3:1:  错误: 找不到引用
[error]  * Prune hive table partitions using partition filters on {@link HiveTableRelation}. The pruned
[error]                                                                  ^
[error] /Users/linxuewei/Code/github/leanken-spark/sql/hive/target/java/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.java:4:1:  错误: 找不到引用
[error]  * partitions will be kept in {@link HiveTableRelation.prunedPartitions}, and the statistics of
[error]                                      ^
[error] /Users/linxuewei/Code/github/leanken-spark/sql/hive/target/java/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.java:9:1:  错误: 找不到引用
[error]  * {@link org.apache.spark.sql.execution.SparkStrategies.JoinSelection}.
[error]           ^
```

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: I was trying to add a empty class definition before object, the unidoc build passed.

```
private[sql] class PruneHiveTablePartitions

/**
 * Prune hive table partitions using partition filters on [[HiveTableRelation]]. The pruned
 * partitions will be kept in [[HiveTableRelation.prunedPartitions]], and the statistics of
 * the hive table relation will be updated based on pruned partitions.
 *
 * This rule is executed in optimization phase, so the statistics can be updated before physical
 * planning, which is useful for some spark strategy, eg.
 * [[org.apache.spark.sql.execution.SparkStrategies.JoinSelection]].
 *
 * TODO: merge this with PruneFileSourcePartitions after we completely make hive as a data source.
 */
private[sql] object PruneHiveTablePartitions
  extends Rule[LogicalPlan] with CastSupport with PredicateHelper {
```

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: So, how do I fix this issue?
1. remove the comment
2. remove the comment with [[]] link
3. add a empty class definition

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 5]
```diff
@@ -255,19 +255,19 @@ class Analyzer(
...
       ResolveBinaryArithmetic ::
       ResolveUnion ::
-      TypeCoercion.typeCoercionRules(conf) ++
+      TypeCoercion.typeCoercionRules ++
       extendedResolutionRules : _*),
     Batch("Post-Hoc Resolution", Once, postHocResolutionRules: _*),
     Batch("Normalize Alter Table", Once, ResolveAlterTableChanges),
     Batch("Remove Unresolved Hints", Once,
-      new ResolveHints.RemoveAllHints(conf)),
+      new ResolveHints.RemoveAllHints),
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:270]: can this be `object`?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 6]
```diff
@@ -771,22 +771,23 @@ class AnalysisSuite extends AnalysisTest with Matchers {
     // RuleExecutor only throw exception or log warning when the rule is supposed to run
     // more than once.
     val maxIterations = 2
-    val conf = new SQLConf().copy(SQLConf.ANALYZER_MAX_ITERATIONS -> maxIterations)
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:774]: do we have to update this test?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala [hunk 7]
```diff
@@ -53,16 +53,18 @@ trait AnalysisTest extends PlanTest {
   }
 
   protected def getAnalyzer(caseSensitive: Boolean) = {
-    if (caseSensitive) caseSensitiveAnalyzer else caseInsensitiveAnalyzer
+    makeAnalyzer(caseSensitive)
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala:56]: why do we need to change it?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinalsSuite.scala [hunk 8]
```diff
@@ -36,31 +37,40 @@ class SubstituteUnresolvedOrdinalsSuite extends AnalysisTest {
...
-      new SubstituteUnresolvedOrdinals(conf).apply(plan),
+      SubstituteUnresolvedOrdinals.apply(plan),
       testRelation2.orderBy(UnresolvedOrdinal(1).asc, UnresolvedOrdinal(2).asc))
 
     // Tests order by ordinal, do full analysis
     checkAnalysis(plan, testRelation2.orderBy(a.asc, b.asc))
 
     // order by ordinal can be turned off by config
+    var newPlan: LogicalPlan = null
+    withSQLConf(SQLConf.ORDER_BY_ORDINAL.key -> "false") {
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinalsSuite.scala:48]: nit:
```
withSQLConf(SQLConf.ORDER_BY_ORDINAL.key -> "false") {
  comparePlans(SubstituteUnresolvedOrdinals.apply(plan), ...)
}
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinalsSuite.scala [hunk 9]
```diff
@@ -36,31 +37,40 @@ class SubstituteUnresolvedOrdinalsSuite extends AnalysisTest {
...
     // Tests group by ordinal, do full analysis
     checkAnalysis(plan2, testRelation2.groupBy(a, b)(a, b))
 
     // group by ordinal can be turned off by config
+    var newPlan2: LogicalPlan = null
+    withSQLConf(SQLConf.GROUP_BY_ORDINAL.key -> "false") {
+      newPlan2 = SubstituteUnresolvedOrdinals.apply(plan2)
+    }
     comparePlans(
-      new SubstituteUnresolvedOrdinals(conf.copy(SQLConf.GROUP_BY_ORDINAL -> false)).apply(plan2),
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinalsSuite.scala:63]: ditto

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala (hunk 6 above)
Comment (leanken) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:774]: after tested, no need to change here anymore. will update.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala [hunk 10]
```diff
@@ -439,7 +438,7 @@ case class RowToColumnarExec(child: SparkPlan) extends RowToColumnarTransition {
   )
 
   override def doExecuteColumnar(): RDD[ColumnarBatch] = {
-    val enableOffHeapColumnVector = sqlContext.conf.offHeapColumnVectorEnabled
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala:442]: The previous code is better, as `sqlContext` is kind of "materialized", which is the one that was used to create the plan node. `conf` always gets the active one.

Comment (leanken) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala:442]: OK. will revert.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala (hunk 7 above)
Comment (leanken) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala:56]: yes, it's duplicated, since caseSensitiveAnalyzer and caseInsensitiveAnalyzer are already change from val to def. will revert.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala (hunk 4 above)
Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: @cloud-fan FYI. need your suggestion too.

File: sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala [hunk 11]
```diff
@@ -51,21 +51,26 @@ class DataSourceAnalysisSuite extends SparkFunSuite with BeforeAndAfterAll {
...
         case _ =>
-          Cast(e, dt, Option(conf.sessionLocalTimeZone))
+          Cast(e, dt, Option(SQLConf.get.sessionLocalTimeZone))
       }
     }
-    val rule = DataSourceAnalysis(conf)
+
+    def rule(caseSensitive: Boolean): DataSourceAnalysis.type =
+    {
+      SQLConf.get.setConf(SQLConf.CASE_SENSITIVE, caseSensitive)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala:65]: we can set conf at the beginning
```
Seq(true, false).foreach { caseSensitive =>
  SQLConf.get.set...
  ...
  val rule = DataSourceAnalysis
}
```

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala (hunk 4 above)
Comment (cloud-fan) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:43]: we can replace [[]] with backticks
```
... using partition filters on `HiveTableRelation`.
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala [hunk 12]
```diff
@@ -53,16 +53,18 @@ trait AnalysisTest extends PlanTest {
...
   }
 
   protected def checkAnalysis(
       inputPlan: LogicalPlan,
       expectedPlan: LogicalPlan,
       caseSensitive: Boolean = true): Unit = {
-    val analyzer = getAnalyzer(caseSensitive)
-    val actualPlan = analyzer.executeAndCheck(inputPlan, new QueryPlanningTracker)
-    comparePlans(actualPlan, expectedPlan)
+    withSQLConf(SQLConf.CASE_SENSITIVE.key -> caseSensitive.toString) {
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala:63]: let's not change the test framework if not necessary. It's OK to test with 2 analyzer instance.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/AggregateOptimizeSuite.scala [hunk 13]
```diff
@@ -29,7 +29,8 @@ import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.internal.SQLConf.{CASE_SENSITIVE, GROUP_BY_ORDINAL}
 
 class AggregateOptimizeSuite extends PlanTest {
-  override val conf = new SQLConf().copy(CASE_SENSITIVE -> false, GROUP_BY_ORDINAL -> false)
+  SQLConf.get.setConf(CASE_SENSITIVE, false)
+  SQLConf.get.setConf(GROUP_BY_ORDINAL, false)
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/AggregateOptimizeSuite.scala:33]: Can we follow https://github.com/apache/spark/pull/30097/files#diff-6f79a980de03203104cfacc6f50c302b843e3b60f1d48942bded6706c75a02b7R128 , instead of updating the global conf object?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/EliminateSortsSuite.scala [hunk 14]
```diff
@@ -32,7 +32,8 @@ import org.apache.spark.sql.internal.SQLConf.{CASE_SENSITIVE, ORDER_BY_ORDINAL}
 import org.apache.spark.sql.types.IntegerType
 
 class EliminateSortsSuite extends PlanTest {
-  override val conf = new SQLConf().copy(CASE_SENSITIVE -> true, ORDER_BY_ORDINAL -> false)
+  SQLConf.get.setConf(CASE_SENSITIVE, true)
+  SQLConf.get.setConf(ORDER_BY_ORDINAL, false)
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/EliminateSortsSuite.scala:36]: ditto

File: sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala [hunk 15]
```diff
@@ -35,6 +35,7 @@ class DataSourceAnalysisSuite extends SparkFunSuite with BeforeAndAfterAll {
 
   override def beforeAll(): Unit = {
     super.beforeAll()
+    SQLConf.get.unsetConf(SQLConf.CASE_SENSITIVE)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala:38]: we should do it in `aferAll`, to avoid this test suite changing the global conf object.

File: sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala [hunk 16]
```diff
@@ -196,6 +202,7 @@ class DataSourceAnalysisSuite extends SparkFunSuite with BeforeAndAfterAll {
     }
 
     test(s"Static partition and dynamic partition (caseSensitive: $caseSensitive)") {
+      SQLConf.get.setConf(SQLConf.CASE_SENSITIVE, caseSensitive)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala:205]: I have a better idea. We can create a `test` method in this suite:
```
def test(caseSensitive: Boolean, testName: String)(func: => Unit) = {
  test(s"$testName (caseSensitive: $caseSensitive)") {
    withSQLConf(case sen....) {
      func
    }
  }
}
```

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala:205]: Then we don't need to change `beforeAll/afterAll`, and it's one line diff for each test.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala [hunk 17]
```diff
@@ -27,7 +27,8 @@ import org.apache.spark.sql.catalyst.planning.PhysicalOperation
 import org.apache.spark.sql.catalyst.plans.logical.{Filter, LogicalPlan, Project}
 import org.apache.spark.sql.catalyst.rules.Rule
 import org.apache.spark.sql.execution.datasources.DataSourceStrategy
-import org.apache.spark.sql.internal.SQLConf
+
+private[sql] class PruneHiveTablePartitions
```
Comment (cloud-fan) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:31]: Why can't we put the comment as classdoc?

Comment (leanken) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala]: done

File: sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala (hunk 16 above)
Comment (leanken) [reply sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceAnalysisSuite.scala]: done

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/EliminateSortsSuite.scala (hunk 14 above)
Comment (leanken) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/EliminateSortsSuite.scala]: done

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/AggregateOptimizeSuite.scala (hunk 13 above)
Comment (leanken) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/AggregateOptimizeSuite.scala]: done

**Comment by leanken** [comment]
retest this, please

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala [hunk 18]
```diff
@@ -243,16 +243,16 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
  * TODO: we should remove the special handling for hive tables after completely making hive as a
  * data source.
  */
-class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan] {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala:246]: I think this rule should still keep the session parameter. The rule should only be dynamic about getting configs, but the session here is used to get the catalog and pass to DS V1.

In practice, this doesn't matter as we always run rules with setting the desired active session. But it's better to be consistent. There are other rules taking catalog as a parameter, e.g. `RelationConversions`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala [hunk 19]
```diff
@@ -73,9 +73,9 @@ class ResolveSQLOnFile(sparkSession: SparkSession) extends Rule[LogicalPlan] {
 /**
  * Preprocess [[CreateTable]], to do some normalization and checking.
  */
-case class PreprocessTableCreation(sparkSession: SparkSession) extends Rule[LogicalPlan] {
+object PreprocessTableCreation extends Rule[LogicalPlan] {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala:76]: ditto

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala [hunk 20]
```diff
@@ -41,7 +41,7 @@ import org.apache.spark.sql.internal.{HiveSerDe, SQLConf}
  * Determine the database, serde/format and schema of the Hive serde table, according to the storage
  * properties.
  */
-class ResolveHiveSerdeTable(session: SparkSession) extends Rule[LogicalPlan] {
+object ResolveHiveSerdeTable extends Rule[LogicalPlan] {
```
Comment (cloud-fan) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala:44]: ditto, this access the catalog: `val dbName = t.identifier.database.getOrElse(session.catalog.currentDatabase)`

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala [hunk 21]
```diff
@@ -40,10 +39,10 @@ import org.apache.spark.sql.internal.SQLConf
  *
  * TODO: merge this with PruneFileSourcePartitions after we completely make hive as a data source.
  */
-private[sql] class PruneHiveTablePartitions(session: SparkSession)
-  extends Rule[LogicalPlan] with CastSupport with PredicateHelper {
+private[sql] class PruneHiveTablePartitions
 
-  override val conf: SQLConf = session.sessionState.conf
+private[sql] object PruneHiveTablePartitions
```
Comment (cloud-fan) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/PruneHiveTablePartitions.scala:44]: ditto
